VERSION-RANGE-SPEC.rst

```diff
@@ -0,0 +1,989 @@
+    and where each version occurs only once in any constraint.
+
+- If the constraint list contains only one item and the comparator is "*",
+  then the "tested version" is IN the range. Check is finished.
+
+- Select the version equality and comparison procedures suitable for this
+  versioning scheme and use these for all version comparisons performed below.
+
+- If the "tested version" is equal to the any of the constraint version
+  where the constraint comparator is for equality (any of "=", "<=", or ">=")
+  then the "tested version" is in the range. Check is finished.
+
+- If the "tested version" is equal to the any of the constraint version where
+  the constraint comparator is "=!" then the "tested version" is NOT in the
+  range. Check is finished.
+
+- Split the constraint list in two sub lists:
+
+  - a first list where the comparator is "=" or "!="
+  - a second list where the comparator is neither "=" nor "!="
+
+- Iterate over the current and next contiguous constraints pairs (aka. pairwise)
+  in the second list.
+
+- For each current and next constraint:
+
+    - If this is the first iteration and current comparator is "<" or <="
+      and the "tested version" is less than the current version
+      then the "tested version" is IN the range. Check is finished.
+
+    - If this is the last iteration and next comparator is ">" or >="
+      and the "tested version" is greater than the next version
+      then the "tested version" is IN the range. Check is finished.
+
+    - If current comparator is ">" or >=" and next comparator is "<" or <="
+      and the "tested version" is greater than the current version
+      and the "tested version" is less than the next version
+      then the "tested version" is IN the range. Check is finished.
+
+    - If current comparator is "<" or <=" and next comparator is ">" or >="
+      then these versions are out the range. Continue to the next iteration.
+
+- Reaching here without having finished the check before means that the
+  "tested version" is NOT in the range.
+
+
+Notes and caveats
+~~~~~~~~~~~~~~~~~~~
+
+- Comparing versions from two different versioning schemes is an error. Even
+  though there may be some similarities between the ``semver`` version of an npm
+  and the ``deb`` version of its Debian packaging, the way versions are compared
+  specific to each versioning scheme and may be different. Tools should report
+  an error in this case.
+
+- All references to sorting or ordering of version constraints means sorting
+  by version. And sorting by versions always implies using the versioning
+  scheme-specified version comparison and ordering.
+
+
+Some of the known versioning schemes
+----------------------------------------
+
+These are a few known versioning schemes for some common Package URL
+`types` (aka. ``ecosystem``).
+
+- **deb**: Debian and Ubuntu https://www.debian.org/doc/debian-policy/ch-relationships.html
+  The comparators are <<, <=, =, >= and >>.
```

Comment:
How will this work? These comparators are not recognized by a vers parser.

Reply:
I should write instead: Debian uses these comparators: <<, <=, =, >= and >>.